samples/powerpoint/shapes/binding-to-shapes.yaml

```diff
@@ -0,0 +1,370 @@
+host: POWERPOINT
+api_set:
+    PowerPointApi: '1.8'
+script:
+    content: |
+        document
+          .getElementById("insert-image-and-create-binding")
+          .addEventListener("click", () => tryCatch(insertImageWithBinding));
+        document.getElementById("replace-binding1").addEventListener("click", () => tryCatch(replaceImageinBinding1));
+        document.getElementById("replace-binding2").addEventListener("click", () => tryCatch(replaceImageinBinding2));
+        document.getElementById("replace-binding3").addEventListener("click", () => tryCatch(replaceImageinBinding3));
+        document.getElementById("load-bindings").addEventListener("click", () => tryCatch(loadBindings));
+        document.getElementById("bring-to-front").addEventListener("click", () => tryCatch(bringToFront));
+        document.getElementById("bring-forward").addEventListener("click", () => tryCatch(bringForward));
+        document.getElementById("send-backward").addEventListener("click", () => tryCatch(sendBackward));
+        document.getElementById("send-to-back").addEventListener("click", () => tryCatch(sendToBack));
+        document.getElementById("setup").addEventListener("click", () => tryCatch(setup));
+
+        var dogImage = getImageAsBase64String();
+
+        var cow1 = returnCowAsBase64();
+
+        var cow2 = returnCow2AsBase64();
+
+        var cow3 = returnCow3AsBase64();
+
+        async function insertImageWithBinding() {
+          // Inserts an image with binding.
+          await PowerPoint.run(async (context) => {
+            const bindingId = (document.getElementById("temp-binding-id") as HTMLInputElement).value;
+            const slide = context.presentation.getSelectedSlides().getItemAt(0);
+            const myShape = slide.shapes.addGeometricShape(PowerPoint.GeometricShapeType.rectangle, {
+              top: 100,
+              left: 30,
+              width: 200,
+              height: 200
+            });
+
+            myShape.fill.setImage(dogImage);
+            context.presentation.bindings.add(myShape, PowerPoint.BindingType.shape, bindingId);
+            await context.sync();
+
+            const bindingsDropdown = document.getElementById("bindings-dropdown") as HTMLSelectElement;
+
+            const option = new Option(`Binding ${bindingId}`, bindingId);
+
+            // When a binding ID already exists, the binding is updated to refer to the new shape
+            // so select the existing item rather than add a new one.
+            const foundIndex = findDropdownItem(bindingsDropdown, option.text);
+            if (foundIndex < 0) {
+              bindingsDropdown.add(option);
+              bindingsDropdown.selectedIndex = bindingsDropdown.options.length - 1;
+            } else {
+              bindingsDropdown.selectedIndex = foundIndex;
+            }
+          });
+        }
+
+        function replaceImageinBinding1() {
```

Comment:
You can drop all 3 of these functions if you use the updated trycatch blocks above.